File: js/helpers/commits.js
import { COMMITS } from "../../data/commits";

export function getCommitFromID(id) {
  return COMMITS[id];
}

export function getCommitIDByHash(hash) {
  let i = 0;
  for (i = 0; i < COMMITS.length; i++) {
    if (COMMITS[i].hash == hash) {
      return i;
    }
  }
  return -1;
}

File: python/commits.py
import json

from helpers import exportJS


def getCommitData(repo):
    data = []
    lookup = {}
    i = 0
    for commit in repo.iter_commits(reverse=True):
        data.append({
            "hash": str(commit),
            "time": commit.committed_date,
            # TODO: "author": commit.author.email,
            "id": i + 1
        })
        lookup[str(commit)] = i
        i += 1
    return (data, lookup)


def writeCommitData(data):
    open("../data/commits.js", "w").write(exportJS({
        "COMMITS": data  #
    }))